### ISSUE TITLE
osx: keyboard events not sent back to application

### ISSUE THREAD

**Issue by paulrouget** [tomaka/winit#131]
My application use cocoa textfield. Because of:

- https://github.com/tomaka/winit/blob/f1d70d351ef9c46ab684d70bd99dcad7d8c9b015/src/platform/macos/mod.rs#L745
- PR: https://github.com/tomaka/glutin/pull/384

… textfields are not editable.

We need to find a better way to address the alert sound issue.

**Comment by paulrouget** [comment]
This appears to solve this issue. I'll do a PR later.

```diff
diff --git a/src/platform/macos/mod.rs b/src/platform/macos/mod.rs
index bd1ed0b6..d3758aba 100644
--- a/src/platform/macos/mod.rs
+++ b/src/platform/macos/mod.rs
@@ -407,7 +407,21 @@ impl Window {
                 appkit::NSFullSizeContentViewWindowMask as NSUInteger
             };

-            let window = IdRef::new(NSWindow::alloc(nil).initWithContentRect_styleMask_backing_defer_(
+
+            use std::sync::{Once, ONCE_INIT};
+            static INIT: Once = ONCE_INIT;
+            INIT.call_once(|| {
+                extern fn on_key_down(_this: &Object, _sel: Sel, _id: id) { }
+                extern fn alloc(_this: &Object, _sel: Sel, _id: id) -> id { unsafe { NSWindow::alloc(nil) } }
+                let superclass = Class::get("NSWindow").unwrap();
+                let mut decl = ClassDecl::new("WinitNSWindow", superclass).unwrap();
+                decl.add_method(sel!(keyDown:), on_key_down as extern fn(&Object, Sel, id));
+                decl.add_method(sel!(alloc:), alloc as extern fn(&Object, Sel, id) -> id);
+                decl.register();
+            });
+            let window: id = msg_send![Class::get("WinitNSWindow").unwrap(), alloc];
+
+            let window = IdRef::new(window.initWithContentRect_styleMask_backing_defer_(
                 frame,
                 masks,
                 appkit::NSBackingStoreBuffered,
```

**Comment by francesca64** [comment]
Fixed via #511
